libraries/chain/controller.cpp

```diff
@@ -386,7 +386,10 @@ struct controller_impl {
          db.undo();
       }
 
-      ilog( "database initialized with hash: ${hash}", ("hash", calculate_integrity_hash()));
+      if( snapshot ) {
+         const auto hash = calculate_integrity_hash();
```

Comment:
Are there side-effects in calculate_integrity_hash that we want to occur even if the information logging is not turned on?

Reply:
No. Should have no side-effects.

Reply:
Nevermind, just read Bart's telegram and I see why we want to do this outside the ilog.